### PR TITLE
Add a test deployment tutorial featuring our CloudFormation template

### LINKED ISSUE #2005: V1 AWS CloudFormation Template: Dev/Test
Background: https://airtable.com/tblD3oZPLJgGhCmch/viwINAgXA5on9pr2T/recz72DK9aRSRPUix

PRs:

### PR DESCRIPTION
**Pull request by jseldess**
Fixes #2005

### CODE REVIEW

**Comment by jseldess** [comment]
@nstewart, PTAL. I'll add the 1.0 and 2.0 versions once I know what revisions are required here.

**Comment by nstewart** [comment]
*[v1.1/deploy-a-test-cluster.md, line 26 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2356#-L2_j7M0X_rmcamdBvDQ:-L2_j7M1QMP1msAcXqxl:bbdju7p) ([raw file](https://github.com/cockroachdb/docs/blob/f47d37ad411881ddb3ebf507208b6a093234c658/v1.1/deploy-a-test-cluster.md#L26)):*
> ```Markdown
>     - For production resiliency, the recommendation would be to span 3 or more availability zones in a single region or 3 or more regions.
> 
> - For a small amount of security, the cluster runs in a virtual private cloud that restricts access to specific IP address ranges. However, the cluster is otherwise insecure. This means that, within the VPC:
> ```

for a small amount of security we CAN restrict access to specific IP address ranges, but the default is open to anyone (0.0.0.0/0)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2356)*

**Comment by nstewart** [comment]
*[v1.1/deploy-a-test-cluster.md, line 17 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2356#-L2_nH4V5J4oU16DX_kW:-L2_nH4V5J4oU16DX_kX:b-e3rhlu) ([raw file](https://github.com/cockroachdb/docs/blob/f47d37ad411881ddb3ebf507208b6a093234c658/v1.1/deploy-a-test-cluster.md#L17)):*
> ```Markdown
> Before getting started, it's important to review some limitations and requirements.
> 
> ### Limitations
> ```

Can we mention something about the various limits that AWS has? Basically in some cases you can only have 20 EC2 instances in a particular region: https://aws.amazon.com/ec2/faqs/#How_many_instances_can_I_run_in_Amazon_EC2

Here's a bigger list of their various quotas: https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2356)*

**Comment by nstewart** [comment]
*[v1.1/deploy-a-test-cluster.md, line 184 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2356#-L2_zZDVtB-D_S7hconi:-L2_zZDVtB-D_S7hconj:b-iaib6l) ([raw file](https://github.com/cockroachdb/docs/blob/f47d37ad411881ddb3ebf507208b6a093234c658/v1.1/deploy-a-test-cluster.md#L184)):*
> ```Markdown
> To see this in action:
> 
> 1. In the terminal where you SSHed into the Kubernetes master node, press **CTRL + C** to stop the `rand` load generator.
> ```

Now is the place where you would actually recommend ssh-ing into the cluster

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2356)*

**Comment by nstewart** [comment]
Review status: 0 of 6 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[v1.1/deploy-a-test-cluster.md, line 49 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2356#-L2_oDy_4wdfvdzdPrdZ:-L2_oDy_4wdfvdzdPrd_:b-imikcn) ([raw file](https://github.com/cockroachdb/docs/blob/f47d37ad411881ddb3ebf507208b6a093234c658/v1.1/deploy-a-test-cluster.md#L49)):*
> ```Markdown
>     The launch process generally takes 10 to 15 minutes. Once you see the `CREATE_COMPLETE` status in the CloudFormation UI, the cluster is ready for testing.
> 
>     {{site.data.alerts.callout_info}}If the launch process times out or fails, you could be running into an <a href="https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html">AWS service limit</a>.{{site.data.alerts.end}}
> ```

You can view any errors in the event history: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-monitor-stack.html

---

*[v1.1/deploy-a-test-cluster.md, line 136 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2356#-L2_ojB-Tka8-GUz6cSF:-L2_ojB-Tka8-GUz6cSG:bxk5vjn) ([raw file](https://github.com/cockroachdb/docs/blob/f47d37ad411881ddb3ebf507208b6a093234c658/v1.1/deploy-a-test-cluster.md#L136)):*
> ```Markdown
> {{site.data.alerts.callout_success}}With the <code>cockroach</code> binary is on the master node, other client <a href="cockroach-commands.html"><code>cockroach</code> commands</a> can be run in the same way.{{site.data.alerts.end}}
> 
> ## Step 3. Start a load generator
> ```

same here -- no need to SSH. 99% of users don't actually need to know K8s is involved. They can treat it as a black box

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2356)*

**Comment by jseldess** [comment]
TFTR, Nate. If you can, please review my changes. I'll create pages for other versions now as well.

---

Review status: 0 of 6 files reviewed at latest revision, 6 unresolved discussions, some commit checks pending.

---

*[v1.1/deploy-a-test-cluster.md, line 17 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2356#-L2_nH4V5J4oU16DX_kW:-L2a9-D1tQgYT3LVp-LC:b-e89n17) ([raw file](https://github.com/cockroachdb/docs/blob/f47d37ad411881ddb3ebf507208b6a093234c658/v1.1/deploy-a-test-cluster.md#L17)):*
<details><summary><i>Previously, nstewart (Nate) wrote…</i></summary><blockquote>

Can we mention something about the various limits that AWS has? Basically in some cases you can only have 20 EC2 instances in a particular region: https://aws.amazon.com/ec2/faqs/#How_many_instances_can_I_run_in_Amazon_EC2

Here's a bigger list of their various quotas: https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html
</blockquote></details>

Given that this is restricted to a single region, and you can't scale beyond 15 nodes, I wonder if going into more detail, somewhat non-applicable limits is helpful. Do you think a sentence following the "15 node` limit would suffice? 

---

*[v1.1/deploy-a-test-cluster.md, line 26 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2356#-L2_j7M0X_rmcamdBvDQ:-L2a9JqmDOtgOu8RB2CS:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f47d37ad411881ddb3ebf507208b6a093234c658/v1.1/deploy-a-test-cluster.md#L26)):*
<details><summary><i>Previously, nstewart (Nate) wrote…</i></summary><blockquote>

for a small amount of security we CAN restrict access to specific IP address ranges, but the default is open to anyone (0.0.0.0/0)
</blockquote></details>

Done.

---

*[v1.1/deploy-a-test-cluster.md, line 49 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2356#-L2_oDy_4wdfvdzdPrdZ:-L2a9af0mkuGwRz3JiwG:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f47d37ad411881ddb3ebf507208b6a093234c658/v1.1/deploy-a-test-cluster.md#L49)):*
<details><summary><i>Previously, nstewart (Nate) wrote…</i></summary><blockquote>

You can view any errors in the event history: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-monitor-stack.html
</blockquote></details>

Done.

---

*[v1.1/deploy-a-test-cluster.md, line 55 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2356#-L2_oSQRJsk2RxNs2hBv:-L2a9bkbjz4MH7imnxnC:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f47d37ad411881ddb3ebf507208b6a093234c658/v1.1/deploy-a-test-cluster.md#L55)):*
<details><summary><i>Previously, nstewart (Nate) wrote…</i></summary><blockquote>

Skip all the SSH proxy commands and just have people run locally. The only reason you need to ssh into the master node is if you want to scale the k8s stateful sets
</blockquote></details>

Done.

---

*[v1.1/deploy-a-test-cluster.md, line 136 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2356#-L2_ojB-Tka8-GUz6cSF:-L2a9cedA18gIl-4pTjo:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f47d37ad411881ddb3ebf507208b6a093234c658/v1.1/deploy-a-test-cluster.md#L136)):*
<details><summary><i>Previously, nstewart (Nate) wrote…</i></summary><blockquote>

same here -- no need to SSH. 99% of users don't actually need to know K8s is involved. They can treat it as a black box
</blockquote></details>

Done.

---

*[v1.1/deploy-a-test-cluster.md, line 184 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2356#-L2_zZDVtB-D_S7hconi:-L2a9e5YLDebtzWI8gUU:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f47d37ad411881ddb3ebf507208b6a093234c658/v1.1/deploy-a-test-cluster.md#L184)):*
<details><summary><i>Previously, nstewart (Nate) wrote…</i></summary><blockquote>

Now is the place where you would actually recommend ssh-ing into the cluster
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2356#-:-L2a9lSxqY39jj6KQ9SU:b-yu2sdy)*

**Comment by nstewart** [comment]
LGTM

File: v1.0/deploy-a-test-cluster.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    - Any user, connecting as `root`, can read or write any data in your cluster.
+
+### Requirements
+
+- You must have an [AWS account](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-sign-up-for-aws.html).
+- You must have [SSH access](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html) in the AWS region where the cluster is deployed.   
+
+## Step 1. Start CockroachDB
+
+1. [Launch the CockroachDB CloudFormation template](http://amzn.to/2C5IFhv).
```
Comment (nstewart) [inline v1.0/deploy-a-test-cluster.md]: Can we use this link instead? This lets us track clicks: http://amzn.to/2CZjJLZ

Comment (jseldess) [reply v1.0/deploy-a-test-cluster.md]: Done.

**Pull request merged**